### ISSUE TITLE
Duplicated collections in orphanCollections after multiple deletions of edge definition.

### ISSUE THREAD

**Issue by garyhai** [arangodb/arangodb#3456]
## my environment running ArangoDB
I'm using the latest ArangoDB of the respective release series:
- [ ] 2.8
- [ ] 3.0
- [ ] 3.1
- [x] 3.2
- [ ] self-compiled devel branch

Mode:
- [ ] Cluster
- [x] Single-Server

Storage-Engine:
- [x] mmfiles
- [ ] rocksdb

On this operating system:
- [ ] DCOS on
  - [ ] AWS
  - [ ] Azure
  - [ ] own infrastructure
- [ ] Linux 
  - [ ] Debian .deb
  - [ ] Ubuntu .deb
  - [ ] SUSE   .rpm
  - [ ] RedHat .rpm
  - [ ] Fedora .rpm
  - [ ] Gentoo
  - [ ] docker - official docker library
  - [ ] other:
- [ ] Windows, version: 
- [x] MacOS, version:

### this is an AQL-related issue:
[ ] I'm using graph features

I'm issuing AQL via:
- [ ] web interface with this browser:     running on this OS:
- [ ] arangosh
- [ ] this Driver:

I've run `db._explain("<my aql query>")` and it didn't shed more light on this.
The AQL query in question is:

The issue can be reproduced using this dataset:

Please provide a way to create the dataset to run the above query on; either by a gist with an arangodump, or `db.collection.save({my: "values"}) statements. If it can be reproduced with one of the ArangoDB example datasets, it's a plus.

### this is a web interface-related issue:
I'm using the web interface with this browser:     running on this OS:
- [ ] authentication is enabled?
- [ ] using the cluster?
- [ ] _system database?

These are the steps to reproduce:
```
shell> curl -X DELETE --dump - http://localhost:8529/_api/gharial/social/edge/relation
shell> curl -X DELETE --dump - http://localhost:8529/_api/gharial/social/edge/relation
shell> curl -X DELETE --dump - http://localhost:8529/_api/gharial/social/edge/relation
```
1) open the browser on http://127.0.0.1:8529
2) log in as ...
3) use database [ ] `_system` [ ] other: 
4) click to ...
...

The following problem occurs: [Screenshot?] 
Instead I would be expecting: 

### this is an installation-related issue:
Describe which steps you carried out, what you expected to happen and what actually happened.

**Comment by maxkernbach** [comment]
Hello, i tried to reproduce your Issue by adding an edge definition with
```
curl -X POST --data-binary @- --dump - http://localhost:8529/_api/gharial/social/edge <<EOF
{ 
  "collection" : "relation", 
  "from" : [ 
    "female", 
    "male" 
  ], 
  "to" : [ 
    "female",
    "male" 
  ] 
}
EOF
```
and removing it multiple times with 
```
curl -X DELETE --dump - http://localhost:8529/_api/gharial/social/edge/relation
```
When accessing orphanCollections using ``graph._orphanCollections();`` in arango shell, no collections were duplicated. Could you provide further information?

**Comment by garyhai** [comment]
```
$ curl -d '{"name":"gtest"}' http://localhost:8529/_api/gharial

{"error":false,"graph":{"name":"gtest","edgeDefinitions":[],"orphanCollections":[],"isSmart":false,"numberOfShards":0,"replicationFactor":1,"smartGraphAttribute":"","_id":"_graphs/gtest","_rev":"_Vy31IHG---"},"code":202}

$ curl -d '{"collection":"knows","from":["persons"],"to":["persons"]}' http://localhost:8529/_api/gharial/gtest/edge

{"error":false,"graph":{"name":"gtest","edgeDefinitions":[{"collection":"knows","from":["persons"],"to":["persons"]}],"orphanCollections":[],"isSmart":false,"numberOfShards":1,"replicationFactor":1,"smartGraphAttribute":"","_id":"_graphs/gtest","_rev":"_Vy31IHG---"},"code":202}

$ curl -X DELETE http://localhost:8529/_api/gharial/gtest/edge/knows

{"error":false,"graph":{"name":"gtest","edgeDefinitions":[],"orphanCollections":["persons"],"isSmart":false,"numberOfShards":1,"replicationFactor":1,"smartGraphAttribute":"","_id":"_graphs/gtest","_rev":"_Vy35tu----"},"code":202}

$ curl -d '{"collection":"knows","from":["persons"],"to":["persons"]}' http://localhost:8529/_api/gharial/gtest/edge
$ curl -X DELETE http://localhost:8529/_api/gharial/gtest/edge/knows
$ curl -d '{"collection":"knows","from":["persons"],"to":["persons"]}' http://localhost:8529/_api/gharial/gtest/edge
$ curl -X DELETE http://localhost:8529/_api/gharial/gtest/edge/knows
... ...
{"error":false,"graph":{"name":"gtest","edgeDefinitions":[],"orphanCollections":["persons","persons","persons","persons"],"isSmart":false,"numberOfShards":1,"replicationFactor":1,"smartGraphAttribute":"","_id":"_graphs/gtest","_rev":"_Vy39qSm---"},"code":202}

**Comment by garyhai** [comment]
<img width="318" alt="screenshot" src="https://user-images.githubusercontent.com/177827/31945300-5c2c26a8-b901-11e7-9fa3-d0f55d303b13.png">

**Comment by maxkernbach** [comment]
Thanks for reporting the bug. When dropping the graph including collections, orphanCollections still exist. Deleting the collection manually with ``graph._removeVertexCollection("persons");`` works (one of the duplicate collections per command removed), but following error occurs: 
```
JavaScript exception in file '/home/max/Git/arangodb/js/client/modules/@arangodb/arangosh.js' at 98,7: ArangoError 1202: document not found
!      throw error;
!      ^
stacktrace: ArangoError: document not found
    at Object.exports.checkRequestResult (/home/max/Git/arangodb/js/client/modules/@arangodb/arangosh.js:96:21)
    at ArangoCollection.update (/home/max/Git/arangodb/js/client/modules/@arangodb/arango-collection.js:1218:12)
    at Graph._removeVertexCollection (/home/max/Git/arangodb/js/common/modules/@arangodb/general-graph.js:1787:16)
    at <shell command>:1:7
```

**Comment by hkernbach** [comment]
Fixed in referenced PRs. 
@maxkernbach Your issue is also gone, because the duplicate error cannot appear anymore.

**Comment by dothebart** [comment]
Hi, 
The fix for this is now available for download with ArangoDB 3.3.19; 
Closing as fixed.
